### DISCUSSION TITLE
More precision about remote changes delay

### DISCUSSION THREAD

**Discussion by darrepac** [abraunegg/onedrive#1405] [Q&A] [answered]
Hi

Can you elaborate more on this: 
`While local changes are uploaded right away, remote changes are delayed until next automated sync cycle when using --monitor`

sync cycle? how often?

thanks

**Comment by abraunegg** [comment, marked as answer]
@darrepac 
The way this client works is as follows:

**Single Sync Operation:**
`--synchronize` method of operation
1. Check OneDrive for changes, download remote changes to local file system
2. Check local disk for changes, upload changes to OneDrive
3. Check OneDrive for any further change, validating that the two locations are in sync
4. Exit

**Ongoing Sync Operation:**
`--monitor` method of operation
1. At application startup, flag local file system entries to be watched with inotify
2. Check OneDrive for changes, download remote changes to local file system
3. Check local disk for changes, upload changes to OneDrive
4. Check OneDrive for any further change, validating that the two locations are in sync
5. Sleep governed by 'monitor_interval' value
6. Repeat 2,3,4,5 until application exit

During the sleep operation, local changes as detected by inotify are uploaded. Changes made on OneDrive are delayed until the next sync  cycle.

Additionally if you use `--help` the following information is provided:
```
--monitor -m
      Keep monitoring for local and remote changes
  --monitor-fullscan-frequency ARG
      Number of sync runs before performing a full local scan of the synced directory
  --monitor-interval ARG
      Number of seconds by which each sync operation is undertaken when idle under monitor mode.
  --monitor-log-frequency ARG
```

Likewise in the usage.md - there is this: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#monitor_interval

**Reply by darrepac** [reply]
Thanks for this (very) detailed answer!

**Reply by zilpe** [reply]
Thanks for this. If I set OneDrive to download only, I'm guessing that it would significantly reduce the `--monitor` cycle time, since it's not checking for local disk changes. Would I be correct in saying this?
@abraunegg

**Reply by abraunegg** [reply]
@ArshZilpe 
> Thanks for this. If I set OneDrive to download only, I'm guessing that it would significantly reduce the --monitor cycle time, since it's not checking for local disk changes. Would I be correct in saying this

This all depends on what you mean by cycle time.

When using `--monitor --download-only` :
```
./onedrive --confdir '~/.config/onedrive-personal/' --monitor --download-only --verbose
Using 'user' Config Dir: /home/alex/.config/onedrive-personal/
Using 'system' Config Dir: 
Configuration file successfully loaded
Notification (dbus) server not available, disabling
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
Application version: v2.4.13-9-g7857a50
Account Type: personal
Default Drive ID: 66d53be8a5056eca
Default Root ID: 66D53BE8A5056ECA!101
Remaining Free Space: 5368709120
Fetching details for OneDrive Root
OneDrive Root exists in the database
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 30
Starting a sync with OneDrive
Applying changes of Path ID: 66D53BE8A5056ECA!101
Updated Remaining Free Space: 5368709120
Syncing this OneDrive Personal Shared Folder: 05Semester
Applying changes of Path ID: BC7D88EC1F539DCF!1443
Sync with OneDrive is complete
Starting a sync with OneDrive
Applying changes of Path ID: 66D53BE8A5056ECA!101
Updated Remaining Free Space: 5368709120
Syncing this OneDrive Personal Shared Folder: 05Semester
Applying changes of Path ID: BC7D88EC1F539DCF!1443
Sync with OneDrive is complete
Starting a sync with OneDrive
Applying changes of Path ID: 66D53BE8A5056ECA!101
Updated Remaining Free Space: 5368709120
Syncing this OneDrive Personal Shared Folder: 05Semester
Applying changes of Path ID: BC7D88EC1F539DCF!1443
Sync with OneDrive is complete
```

As you can see, it is only ever checking data online, never the local disk, and any local disk changes are ignored, but the time to process changes between `Starting a sync with OneDrive` and `Sync with OneDrive is complete` depends totally on number of items that have changed, and size of items to download.

The `OneDrive monitor interval (seconds): 30` (the wait time between being complete and starting a new check), the sleep is governed by 'monitor_interval' value remains the same - in my example it is 30 seconds, but this is far too frequent but is used for illustrative purposes. The 'monitor_interval' value should be set based on the rate of data change online - if your data online is only changing every hour or every day, why check it every 300 seconds ...
